tendermint/src/abci/types.rs

```diff
@@ -22,13 +23,55 @@ pub struct Validator {
 
 /// Information about a whether a validator signed the last block.
 ///
-/// [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#voteinfo)
+/// [ABCI documentation](https://github.com/cometbft/cometbft/blob/v0.38.x/spec/abci/abci%2B%2B_methods.md#voteinfo)
 #[derive(Clone, PartialEq, Eq, Debug)]
 pub struct VoteInfo {
     /// Identifies the validator.
     pub validator: Validator,
     /// Whether or not the validator signed the last block.
-    pub signed_last_block: bool,
+    pub sig_info: BlockSignatureInfo,
+}
+
+/// Information about a whether a validator signed the last block,
+/// together with vote extensions.
+///
+/// [ABCI documentation](https://github.com/cometbft/cometbft/blob/v0.38.x/spec/abci/abci%2B%2B_methods.md#extendedvoteinfo)
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ExtendedVoteInfo {
+    /// Identifies the validator.
+    pub validator: Validator,
+    /// Whether or not the validator signed the last block.
+    pub sig_info: BlockSignatureInfo,
+    /// Non-deterministic extension provided by the sending validator's application.
+    pub vote_extension: Bytes,
+    /// Signature for the vote extension.
+    pub extension_signature: Option<Signature>,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Information on how the validator voted for a block.
+pub enum BlockSignatureInfo {
+    /// Full information available, as determined by the `BlockIdFlag` value.
+    Flag(block::BlockIdFlag),
+    /// In CometBFT versions before 0.38, the `signed_last_block` field has
+    /// the value of true.
+    ///
+    /// This variant should not be used in CometBFT 0.38 or later
+    /// and will result in the "undefined" encoding in protobuf.
+    LegacySigned,
+}
```

Comment:
@sergio-mena This is my attempt to represent the block signing information in `VoteInfo` and `ExtendedVoteInfo` without data loss also when converted from older versions.
Something to consider in connection with https://github.com/cometbft/cometbft/pull/780

Reply:
I see a problem here. How about the case where we have the "legacy" field set to false?
If these data structures are to work for, e.g., legacy-to-legacy interoperability, you are not able to handle the case where the signature is absent.